src/MissionManager/MissionController.cc

```diff
@@ -754,20 +806,54 @@ void MissionController::_recalcAltitudeRangeBearing()
             if (!item->isStandaloneCoordinate()) {
                 firstCoordinateItem = false;
                 if (lastCoordinateItem != homeItem || (showHomePosition && linkBackToHome)) {
-                    double azimuth, distance, altDifference;
+                    double azimuth, distance, altDifference, telemetryDistance;
 
                     // Subsequent coordinate items link to last coordinate item. If the last coordinate item
                     // is an invalid home position we skip the line
                     _calcPrevWaypointValues(homePositionAltitude, item, lastCoordinateItem, &azimuth, &distance, &altDifference);
                     item->setAltDifference(altDifference);
                     item->setAzimuth(azimuth);
                     item->setDistance(distance);
+
+                    missionDistance += distance;
+
+                    if (item->isSimpleItem()) {
+                        _calcHomeDist(item, homeItem, &telemetryDistance);
+
+                        if (vtolCalc) {
+                            SimpleMissionItem* simpleItem = qobject_cast<SimpleMissionItem*>(item);
+                            if (simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_TAKEOFF || hoverDistanceCalc){
+                                hoverDistance += distance;
+                            }
+                            cruiseDistance = missionDistance - hoverDistance;
+                            if(simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_LAND && !linkBackToHome && !cruiseTransition && !hoverTransition){
+                                hoverDistance = cruiseDistance;
+                                cruiseDistance = missionDistance - hoverDistance;
+                            }
+                        }
+                    } else {
+                        missionDistance += qobject_cast<ComplexMissionItem*>(item)->surveyDistance();
```

Comment:
very good ! - thank you, I failed to discover it in settings.
As for the "survey not included"
I have recent Android build, so I'll describe it:
-start application
-enter planning, click survey icon
-click "draw polygon"
-draw a polygon
-click "finish draw"
=Observe that the Total mission is =m, 0s, max telemetry=0m
-add a waypoint, and there will be time, distance.
-delete the WP, and the time/distance is back to 0.

On 2 August 2016 at 18:30, Adyasha Dash <notifications@github.com> wrote:

> In src/MissionManager/MissionController.cc
> <https://github.com/mavlink/qgroundcontrol/pull/3863#discussion_r73189873>
> :
>
> > +                    if (item->isSimpleItem()) {
> > +                        _calcHomeDist(item, homeItem, &telemetryDistance);
> > +
> > +                        if (vtolCalc) {
> > +                            SimpleMissionItem* simpleItem = qobject_cast<SimpleMissionItem*>(item);
> > +                            if (simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_TAKEOFF || hoverDistanceCalc){
> > +                                hoverDistance += distance;
> > +                            }
> > +                            cruiseDistance = missionDistance - hoverDistance;
> > +                            if(simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_LAND && !linkBackToHome && !cruiseTransition && !hoverTransition){
> > +                                hoverDistance = cruiseDistance;
> > +                                cruiseDistance = missionDistance - hoverDistance;
> > +                            }
> > +                        }
> > +                    } else {
> > +                        missionDistance += qobject_cast<ComplexMissionItem*>(item)->surveyDistance();
>
> @AndKe <https://github.com/AndKe> Survey Distance is taken into account
> here, I think.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mavlink/qgroundcontrol/pull/3863/files/c10431fe5eb3964d472e281989da87fd14fb8b1f#r73189873>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZpqk_laxvUITSPv6rpeI4wPaM2dyjBks5qb3CrgaJpZM4JVLu0>
> .
>